Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,655 @@
+        QTestApi.TestrunApi testrunApi = new QTestApi.TestrunApi();
+        QTestApi.TestcaseApi testcaseApi = new QTestApi.TestcaseApi();
+        QTestApi.FieldApi fieldApi = new QTestApi.FieldApi();
+        QTestApi.TestlogApi testlogApi = new QTestApi.TestlogApi();
+
+        QTestApiClient.ApiClient apiClient = new QTestApiClient.ApiClient();
+        QTestApiClient.Configuration configuration = new QTestApiClient.Configuration();
+
+        public override bool ConnectALMServer()
+        {
+            try
+            {
+                System.Diagnostics.Trace.WriteLine("Initiated Authentication");
+
+                connObj = new QTestApi.LoginApi(ALMCore.DefaultAlmConfig.ALMServerURL);
+                string granttype = "password";
+                string authorization = "Basic bWFoZXNoLmthbGUzQHQtbW9iaWxlLmNvbTo=";
+                QTestApiModel.OAuthResponse response = connObj.PostAccessToken(granttype, ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword, authorization);
+                connObj.Configuration.MyAPIConfig = new QTestApiClient.QTestClientConfig();
+                connObj.Configuration.AccessToken = response.AccessToken;
+                connObj.Configuration.ApiKey.Add("Authorization", response.AccessToken);
+                connObj.Configuration.ApiKeyPrefix.Add("Authorization", response.TokenType);
+                System.Diagnostics.Trace.WriteLine("Authentication Successful");
+                return true;
+            }           
+            catch (Exception ex)
+            {
+                System.Diagnostics.Trace.WriteLine("Exception in AuthenticateUser(): Authentication Failed: " + ex.Message);
+
+                connObj = null;
+                return false;
+            }
+        }
+
+        public override bool ConnectALMProject()
+        {
+            ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
+            return true;
+        }
+
+        public override Boolean IsServerConnected()
+        {
+            // return QCConnect.IsServerConnected;
+            return true;
+        }
+
+        public override void DisconnectALMServer()
+        {
+            // QCConnect.DisconnectQCServer();
+        }
+
+        public override List<string> GetALMDomains()
+        {
+            // return QCConnect.GetQCDomains();
+            return new List<string>();
+        }
+
+        public override Dictionary<string,string> GetALMDomainProjects(string ALMDomain)
+        {
+            projectsApi = new QTestApi.ProjectApi(connObj.Configuration);
+            List<QTestApiModel.ProjectResource> projectList =  projectsApi.GetProjects("descendents", true);
+            return projectList.ToDictionary(f => f.Id.ToString(), f => f.Name);
+        }
+
+        public override bool DisconnectALMProjectStayLoggedIn()
+        {
+            // return QCConnect.DisconnectQCProjectStayLoggedIn();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)